### ISSUE TITLE
T5 Model Bringup

### ISSUE THREAD

**Issue by nsmithtt** [tenstorrent/tt-mlir#75]
Unique ops in T5

- Eltwise
  - multiply
  - add
  - sqrt
  - subtract
  - recip
  - exp
- Reduce Avg
- Reduce Max
- Reduce Sum
- Reshape
- Matmul
- hslice
- hstack
- transpose

**Comment by nvukobratTT** [comment]
Besides, the proposed set of ops, these are related to the inference? I assume training for T5 is still TBD until we get few more PoC for the basic MNIST models working e2e.

**Comment by nsmithtt** [comment]
> What do you think about creating milestones for each model that we're bringing up?
> 
> As we're at the early stage of this compiler, there will be more jobs for each model to work e2e. Therefore, it seems reasonable to have milestones for each one until we see that the new model bringup is becoming less complex.
> 
> I'm asking this as a new op bringup can be a good ramp-up task for new folks on Forge and MLIR projects. Therefore, having some of them in the backlog is beneficial.

I agree completely and I really like this idea of having milestones for each model.  There will be a lot of overlapping work between the models so it could help to have individuals run ahead, try pushing on a model, file issues for the ops/issues they're running into and then we can cross-reference which models are blocked on what which can help us prioritize what the best bang for buck is.

And yet, above was just inference.

**Comment by staylorTT** [comment]
@nvukobratTT Do we still want to keep these issues over with the change of how we're approaching generality?

**Comment by nvukobratTT** [comment]
We're going to keep them on FFE, but no need to keep them on MLIR anymore.
